service/history/workflow/mutable_state_impl.go

```diff
@@ -4558,17 +4556,24 @@ func (ms *MutableStateImpl) closeTransactionHandleWorkflowReset(
 func (ms *MutableStateImpl) closeTransactionHandleActivityUserTimerTasks(
 	transactionPolicy TransactionPolicy,
 ) error {
-
-	if transactionPolicy == TransactionPolicyPassive ||
-		!ms.IsWorkflowExecutionRunning() {
+	switch transactionPolicy {
+	case TransactionPolicyActive:
+		if !ms.IsWorkflowExecutionRunning() {
+			return nil
+		}
+		if err := ms.taskGenerator.GenerateActivityTimerTasks(); err != nil {
+			return err
+		}
+		return ms.taskGenerator.GenerateUserTimerTasks()
+	case TransactionPolicyPassive:
 		return nil
+	default:
+		panic(fmt.Sprintf("unknown transaction policy: %v", transactionPolicy))
```

Comment:
return an internal error?

Reply:
it is better to fail fast here, e.g. panic

i would consider return error when
* input validation fails
* modifying a complicated component & do not want to break production

there are only 2 policies here, no safety concern